cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -188,11 +191,14 @@ public SQLDataset pushInternal(String datasetName,
 
     Runnable pullTask = () -> {
       try {
+        LOG.debug("Starting pull for dataset '{}'", job.getDatasetName());
         waitForJobAndHandleExceptionInternal(job);
         JavaRDD<T> result = pullInternal(job.waitFor(), jsc);
+        LOG.debug("Completed pull for dataset '{}'", job.getDatasetName());
         future.complete(result);
-      } catch (Exception e) {
-        future.completeExceptionally(e);
+      } catch (Throwable t) {
+        LOG.warn("Error when executing pull for dataset '{}'", job.getDatasetName());
```

Comment:
This does not tells which error. Also it duplicate all the logging in it `wait`. Could you add throwable and make one of it DEBUG?

Reply:
Ah, good catch. I'll remove these logs